fs.c

```diff
@@ -1792,7 +1792,7 @@ int FS_ZipUnpackToTemp (unzFile zip_file,
 	int	retval = UNZ_OK;
 
 	// Get a unique temp filename.
-	if (!COM_GetUniqueTempFilename (NULL, unpack_path, unpack_path_size, true))
+	if (!COM_GetUniqueTempFilename (NULL, unpack_path, unpack_path_size, true) < 0)
```

Comment:
`COM_GetUniqueTempFilename` returns `-1` on error and `strlen(filename)` otherwise.

Get rid of the `!` here. `!(-1)` becomes `0`. Then you compare `0 < 0`. So you never enter this `if` statement.

Reply:
Absolutely true! For some reason I only managed to do that to one call site.

Code has been updated.